### ISSUE TITLE
The latest commit in 2.0.0 is causing "fatal error: concurrent map iteration and map write"

### ISSUE THREAD

**Issue by ViperelB** [JakeWnuk/maskcat#1]
fatal error: concurrent map iteration and map write

```
goroutine 70 [running]:
runtime.dopanic__m
        ../../../src/libgo/go/runtime/panic.go:1204
runtime.fatalthrow
        ../../../src/libgo/go/runtime/panic.go:1073
runtime.throw
        ../../../src/libgo/go/runtime/panic.go:1044
runtime.mapiternext
        ../../../src/libgo/go/runtime/map.go:929
github.com/jakewnuk/maskcat/internal/cli.MutateMasks..func1
        /home/kali/maskcat/internal/cli/cli.go:196
runtime.kickoff
        ../../../src/libgo/go/runtime/proc.go:1316
created by github_0com_1jakewnuk_1maskcat_1internal_1cli.MutateMasks
        /home/kali/maskcat/internal/cli/cli.go:193 +0x40a
```

**Comment by JakeWnuk** [comment]
Thanks for submitting this issue!

Did a late night commit and will look into this and issue a fix shortly.

**Comment by JakeWnuk** [comment]
I was able to reproduce this issue and issued a fix in [7603312](https://github.com/JakeWnuk/maskcat/commit/76033122882d64439449871c3e32a551d2cd98ae). Please let me know if this resolved the issue.

Additionally, note that the new logic is supported by concurrent goroutines now. In the past this caused some speed issues so I will be keeping an eye on average performance but any user reports of speed issues would be beneficial.

**Comment by JakeWnuk** [comment]
Going to go ahead and close this issue. Please reach out if there is any more issues.
